pkg/processor/metricfrequencyprocessor/metriccache.go

```diff
@@ -0,0 +1,55 @@
+package metricfrequencyprocessor
+
+import (
+	"github.com/patrickmn/go-cache"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"time"
+)
+
+type DataPoint struct {
+	Timestamp pdata.Timestamp
+	Value     float64
+}
+
+type MetricCache struct {
+	internalCaches map[string]*cache.Cache
+}
+
+func (mc *MetricCache) Register(name string, dataPoint pdata.DoubleDataPoint) {
+
+	internalCache, found := mc.internalCaches[name]
+	if !found {
+		newCache := mc.newCache()
+		mc.internalCaches[name] = newCache
+		internalCache = newCache
+	}
+
+	key := dataPoint.Timestamp().String()
+	value := &DataPoint{Timestamp: dataPoint.Timestamp(), Value: dataPoint.Value()}
+	internalCache.Set(key, value, cache.DefaultExpiration)
+}
+
+func (mc *MetricCache) List(metricName string) map[pdata.Timestamp]float64 {
+	out := make(map[pdata.Timestamp]float64)
+	internalCache, found := mc.internalCaches[metricName]
+	if found {
+		for _, item := range internalCache.Items() {
+			dataPoint := item.Object.(*DataPoint)
```

Comment:
```suggestion
			dataPoint, ok := item.Object.(*DataPoint)
			if !ok {
				...
			}
```

Reply:
I understand defensive programming, but what exactly can we do with `!ok` here? Panic?

My code follows library docs: https://pkg.go.dev/github.com/patrickmn/go-cache#Cache.Add

Reply:
You can propagate an error upstream and check it there

Reply:
There is no error in this api, it's just lack of generics in the library.